src/refiners/foundationals/latent_diffusion/schedulers/euler.py

```diff
@@ -22,31 +24,45 @@ def __init__(
             device=device,
             dtype=dtype,
         )
-        self.sigmas = self.noise_std / self.cumulative_scale_factors
-        self.sigmas[0] = 0.0
+        self.sigmas = self._generate_sigmas().to(device=self.device, dtype=self.dtype)
 
     def _generate_timesteps(self) -> Tensor:
+        # using "trailing" timestep
         step_ratio = self.num_train_timesteps / self.num_inference_steps
         timesteps = arange(1000, 0, -step_ratio).round().type(int64)
         return timesteps - 1
 
-    def __call__(self, x: Tensor, noise: Tensor, step: int) -> Tensor:
-        timestep, previous_timestep = (
-            self.timesteps[step],
-            (
-                self.timesteps[step + 1]
-                if step < self.num_inference_steps - 1
-                else tensor(data=[0], device=self.device, dtype=self.dtype)
-            ),
-        )
-        current_sigma, previous_sigma = self.sigmas[timestep], (
-            self.sigmas[previous_timestep] if previous_timestep > 0 else self.sigmas[0]
+    def _generate_sigmas(self) -> Tensor:
+        sigmas = self.noise_std / self.cumulative_scale_factors
+        sigmas = torch.from_numpy(
+            np.interp(self.timesteps.cpu().numpy(), np.arange(0, len(sigmas)), sigmas.cpu().numpy())
         )
+        sigmas = torch.cat([sigmas, tensor([0.0])])
+        print(sigmas)
+        return sigmas
 
-        predicted_x = x - current_sigma * noise
+    def __call__(
+        self,
+        x: Tensor,
```

Comment:
There are two other specific parts for Euler which are missing in this PR:

1. Before the denoising loop, the initial latents are rescaled - [example](https://github.com/huggingface/diffusers/blob/08b453e3828f80027d881bb460716af95e192bcd/src/diffusers/schedulers/scheduling_euler_discrete.py#L241-L248) in diffusers
2. For each denoising step, the current latents passed to the UNet are also rescaled - [example](https://github.com/huggingface/diffusers/blob/08b453e3828f80027d881bb460716af95e192bcd/src/diffusers/schedulers/scheduling_euler_discrete.py#L257-L281) in diffusers

It's true that the other schedulers currently implemented in Refiners do not require such re-scaling. So we'll need ad hoc mechanisms and/or `Scheduler` evolutions cc @limiteinductive